index.js

```diff
@@ -84,14 +89,17 @@ function _walkSync(baseDir, options, _relativePath) {
 
   for (var i=0; i<sortedEntries.length; ++i) {
     var entry = sortedEntries[i];
+    if (ignoreMatcher && ignoreMatcher.match(entry.relativePath)) {
```

Comment:
Although correct, we may want to do this check between line 67 and 68. This prevents us from doing costly the costly `getStat`.